book/plugins/how-to-guides/maximize-compatibility.md

```diff
@@ -18,13 +18,163 @@ Some guidance on each of these approaches:
    - Your install instructions can indicate that a user should install whichever distribution you depend on (e.g., `tiny`, `amplicon`, or `metagenome`) and then illustrate how to install your plugin(s) in that environment however it makes sense (e.g., `conda` or `pip`). Complete install instructions are drafted for you in the `README.md` of plugins that you build using our template (see [](plugin-from-template)).
    - Alternatively, you can compose and share your own distribution of plugins (e.g., building from the `tiny` distribution) that captures the set of functionality you’d like to share.
 
-Either of these approaches is totally fine.
-The former is an easier starting point.
+Either of these approaches is totally fine, with the following caveat.
+The former is an easier starting point, and will allow us to provide more troubleshooting assistance for any installation issues that users may encounter. While the latter provides you with more flexibility in your environment construction, our assistance with any install issues or environment conflicts that users may run into will be more limited (and will ultimately be your responsibility to troubleshoot and resolve).
+
+With the above information in mind, below are examples of how we recommend constructing installation instructions for users for both of the discussed pathways.
+
+1. Installing your plugin on top of an existing QIIME 2 Distribution (recommended)
+
+The easiest way to instruct users to install your plugin in the context of an existing QIIME 2 Distribution is to create an environment file that they will use to install the relevant distribution along with the correct version of your plugin, all while using a single command.
+
+In the top-level directory of your plugin, create the following directory:
+
+```
+environment-files/
+```
+
+Within this directory, you'll create environment file(s) for current and/or past installable versions of your plugin. You can name them something like:
+
+```
+20XX.REL-<your-plugin-name>-environment.yml
+```
+
+The contents of your environment file should look something like this:
+
+```
+channels:
+- https://packages.qiime2.org/qiime2/20XX.REL/<target-distribution>/passed
+- conda-forge
+- bioconda
+- defaults
+dependencies:
+  - qiime2-<target-distribution>
+  - pip
+  - pip:
+    - <q2_my_plugin>@git+https://github.com/<your-github-org>/<q2-my-plugin>.git@<target-branch>
+```
+
+With the following terms defined:
+- 20XX.REL epoch corresponding to the QIIME 2 release version that your plugin is compatible with
+- <target-distribution> being the distribution your plugin should be utilized with (i.e. amplicon, metagenome)
+- <q2_my_plugin> being the name of your plugin
```

Comment:
I've been calling this the `module_name` in the cookiecutter (see [here](https://github.com/caporaso-lab/cookiecutter-qiime2-plugin/blob/main/cookiecutter.json)) - unless you think there's a reason not to, can you use that term here too for consistency?

Reply:
yep, can do!